src/averageAges.js

```diff
@@ -37,7 +42,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+    && people.some(child => (person.name === child.mother)))
+    : people.filter(person => person.sex === 'f');
+
+  return women.reduce((a, b) => a + (b.died - b.born), 0) / women.length;
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.

Also as an option(up to you) you could create 2 separate variables: `women` and `womenWithChildren` and pass them as arguments to call your new `getAverage` function. This way ternary operator would be in return statement and code would be more clear.